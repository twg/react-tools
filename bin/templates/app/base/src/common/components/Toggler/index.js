import React from 'react'
import PropTypes from 'prop-types'
import css from './style.scss'

class Toggler extends React.Component {
  constructor(props) {
    super(props)
    this.state = { show: false }
    this.toggle = this.toggle.bind(this)
    this.toggleText = this.toggleText.bind(this)
  }

  toggle() {
    this.setState({ show: !this.state.show })
  }

  toggleText() {
    return { __html: this.state.show ? '&uarr;' : '&darr;' }
  }

  render() {
    return (
      <div>
        <div>
          {this.props.children} |{' '}
          <span
            className={css.toggler}
            onClick={this.toggle}
            dangerouslySetInnerHTML={this.toggleText()}
          />
        </div>
        <div>{this.state.show && this.props.more}</div>
      </div>
    )
  }
}

Toggler.propTypes = {
  more: PropTypes.node,
  children: PropTypes.string
}

export default Toggler
