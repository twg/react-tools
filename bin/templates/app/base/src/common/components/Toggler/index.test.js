/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Toggler from './'

describe('Toggler', () => {
  it('matches snapshot', () => {
    const toggler = <Toggler more="more">Text</Toggler>
    let wrapper = renderer.create(toggler).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
