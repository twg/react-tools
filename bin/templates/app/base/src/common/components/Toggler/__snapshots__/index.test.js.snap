// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggler matches snapshot 1`] = `
<div>
  <div>
    Text
     | 
    <span
      className="toggler"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&darr;",
        }
      }
      onClick={[Function]}
    />
  </div>
  <div />
</div>
`;
